builder/dockerfile/evaluator.go

```diff
@@ -194,12 +236,14 @@ type dispatchState struct {
 	imageID    string
 	baseImage  builder.Image
 	stageName  string
+	shlex      *ShellLex
+	builder    *Builder
+	source     builder.Source
```

Comment:
I was thinking of creating a dispatcher struct like
```type dispatcher struct {
 state dispatcherState
 shlex *ShellLex
 builder *Builder
 source builder.Source
}
``` 
and attach current dispatch functions as methods of this struct (that would avoid creating dispatch requests and simplify their signature).